<launch>

	<machine name="Panda00" address="LabRob-Panda-00" user="panda" default="false" env-loader="/home/panda/rosenv.sh" timeout="180" /> -->
	<machine name="Panda01" address="LabRob-Panda-01" user="panda" default="false" env-loader="/home/panda/rosenv.sh" timeout="180" />
	<machine name="Panda02" address="LabRob-Panda-02" user="panda" default="false" env-loader="/home/panda/rosenv.sh" timeout="180" />
	<machine name="Panda03" address="LabRob-Panda-03" user="panda" default="false" env-loader="/home/panda/rosenv.sh" timeout="180" /> -->


	<!-- send Saetta urdf to param server -->
	<param name="saetta_vision_prefix" value="robot_"/>
	<param name="saetta_vision_robnum" value="3" />
	<param name="saetta_vision_cmd_topic" value="cmd_vel" />



	<!-- == ROBOT 01 == -->
		<param name="robot_1_to_master/master" value="true"/>
		<param name="robot_1_to_master/filter" value="5"/>
		<param name="robot_1_to_master/delay" value="0.25"/>
		<node pkg="timestats" type="ts_generator" name="robot_1_to_master">
			<remap from="input" to="robot_1_to_master_end"/>
			<remap from="output" to="robot_1_to_master_start"/>
		</node>

		<node pkg="timestats" type="ts_generator" name="robot_1_to_master_link" machine="Panda01">
			<remap from="input" to="robot_1_to_master_start"/>
			<remap from="output" to="robot_1_to_master_end"/>
		</node>


		<group ns="robot_1">
		<param name="tf_prefix" value="robot_1"/>
		<param name="robot_description" command="$(find xacro)/xacro.py $(find saetta_base)/Saetta.urdf" />
		
		<!-- Robot State Publisher -->
		<node pkg="robot_state_publisher" type="state_publisher" name="state_publisher"/>		

		<!-- Saetta Nodes -->
		<node pkg="saetta_base" type="base" name="SaettaBase" machine="Panda01">
			<remap from="/saetta/velocity" to="cmd_vel"/>
		</node>

		<node pkg="tf" type="static_transform_publisher" name="tf_static" args="0.1 0 0.1 0 0 0 body laser 10" />"
		</group>
	<!-- == END ROBOT 01 == -->


	<!-- == ROBOT 02 == -->
		<param name="robot_2_to_master/master" value="true"/>
		<param name="robot_2_to_master/filter" value="5"/>
		<param name="robot_2_to_master/delay" value="0.25"/>
		<node pkg="timestats" type="ts_generator" name="robot_2_to_master">
			<remap from="input" to="robot_2_to_master_end"/>
			<remap from="output" to="robot_2_to_master_start"/>
		</node>

		<node pkg="timestats" type="ts_generator" name="robot_2_to_master_link" machine="Panda02">
			<remap from="input" to="robot_2_to_master_start"/>
			<remap from="output" to="robot_2_to_master_end"/>
		</node>
		<group ns="robot_2">
		<param name="tf_prefix" value="robot_2"/>
		<param name="robot_description" command="$(find xacro)/xacro.py $(find saetta_base)/Saetta.urdf" />
		
		<!-- Robot State Publisher -->
		<node pkg="robot_state_publisher" type="state_publisher" name="state_publisher"/>		

		<!-- Saetta Nodes -->
		<node pkg="saetta_base" type="base" name="SaettaBase" machine="Panda02">
			<remap from="/saetta/velocity" to="cmd_vel"/>
		</node>

		<node pkg="tf" type="static_transform_publisher" name="tf_static" args="0.1 0 0.1 0 0 0 body laser 10" />"
		</group>
	<!-- == END ROBOT 02 == -->


	<!-- == ROBOT 03 == -->
		<param name="robot_3_to_master/master" value="true"/>
		<param name="robot_3_to_master/filter" value="5"/>
		<param name="robot_3_to_master/delay" value="0.25"/>
		<node pkg="timestats" type="ts_generator" name="robot_3_to_master">
			<remap from="input" to="robot_3_to_master_end"/>
			<remap from="output" to="robot_3_to_master_start"/>
		</node>

		<node pkg="timestats" type="ts_generator" name="robot_3_to_master_link" machine="Panda03">
			<remap from="input" to="robot_3_to_master_start"/>
			<remap from="output" to="robot_3_to_master_end"/>
		</node>
		<group ns="robot_3">
		<param name="tf_prefix" value="robot_3"/>
		<param name="robot_description" command="$(find xacro)/xacro.py $(find saetta_base)/Saetta.urdf" />
		
		<!-- Robot State Publisher -->
		<node pkg="robot_state_publisher" type="state_publisher" name="state_publisher"/>		

		<!-- Saetta Nodes -->
		<node pkg="saetta_base" type="base" name="SaettaBase" machine="Panda03">
			<remap from="/saetta/velocity" to="cmd_vel"/>
		</node>

		<node pkg="tf" type="static_transform_publisher" name="tf_static" args="0.1 0 0.1 0 0 0 body laser 10" />"
		</group>
	<!-- == END ROBOT 03 == -->




</launch>
